packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -284,12 +284,84 @@ class NestedScrollView extends StatefulWidget {
   }
 
   @override
-  _NestedScrollViewState createState() => _NestedScrollViewState();
+  NestedScrollViewState createState() => NestedScrollViewState();
 }
 
-class _NestedScrollViewState extends State<NestedScrollView> {
+/// The [State] for a [NestedScrollView].
+///
+/// It allows you to obtain the [ScrollController]s of the children of your [NestedScrollView].
+///
+/// You can access the inner scroll controller via [innerController] and the outer
+/// controller with [outerController], however, notice that the outer scroll position
+/// can also be obtained using [NestedScrollView.controller].
+///
+/// If you want to access the inner scroll controller of your nested scroll view,
+/// you can get its [NestedScrollViewState] by supplying a `GlobalKey<NestedScrollViewState>`
+/// to your [NestedScrollView] widget (the `key` parameter).
+///
+/// {@tool sample}
+///
+/// You would access the [NestedScrollViewState] using a [GlobalKey].
+/// Using the following setup, you can access the inner scroll controller
+/// using `globalKey.currentState.innerController`.
+///
+/// ```dart
+/// class ExampleWidget extends StatelessWidget {
+///   final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return NestedScrollView(
+///       key: globalKey,
+///       headerSliverBuilder: (context, _) => const [SliverAppBar()],
+///       body: const CustomScrollView(
+///         // ...
+///       ),
+///     );
+///   }
+///
+///   ScrollController get innerController {
+///     return globalKey.currentState.innerController;
+///   }
+/// }
+/// ```
+///
+/// {@end-tool}
+///
+/// Alternatively, you could also visit child elements of your parent widget to obtain
+/// the state if you know exactly that the widget is a direct descendant.
+class NestedScrollViewState extends State<NestedScrollView> {
   final SliverOverlapAbsorberHandle _absorberHandle = SliverOverlapAbsorberHandle();
 
+  /// The [ScrollController] provided to the [ScrollView] in [NestedScrollView.body].
+  ///
+  /// Manipulating the [ScrollPosition] of this controller fully pushes the header sliver down,
```

Comment:
What do you mean by "pushes the header sliver down"? This paragraph is not very clear...

Reply:
I tried making it a little clearer. Do you have an idea what else I could write? I also noticed that the statement was false before, so I corrected it.